### PR TITLE
Feature/v2 storage

### PR DESCRIPTION
**Pull request by SuddenGunter**
- add sync from monobank
- add postgresql storage layer

### CODE REVIEW

File: mono/transaction.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	transactions := make([]Transaction, 0)
+	if err = json.Unmarshal(response, &transactions); err != nil {
+		return nil, fmt.Errorf("failed unmarshal transactions form json body: %w", err)
+	}
+
+	return transactions, nil
+}
+
+// todo come up with better naming
```
Comment (flerka) [inline mono/transaction.go:52]: remove todo if not valid

File: storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	return tx.Commit(ctx)
+}
+
+// GetLastTransactionDate returns date property of latest transaction (sorted by date desc).
+func (s *PostgreSQLStorage) GetLastTransactionDate(ctx context.Context, accountID string) (time.Time, error) {
+	row := s.pool.QueryRow(
+		ctx,
+		`select "time" from transaction
+		where accountID = $1
+		order by time desc
```
Comment (flerka) [inline storage/storage.go:63]: maybe time should be in quotes?

File: storage/storage.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		if err == pgx.ErrNoRows {
+			return time.Time{}, ErrNotFound
+		}
+		return time.Time{}, err
+	}
+
+	return lastKnownTransaction, nil
+}
+
+var ErrNotFound = errors.New("data not found")
```
Comment (flerka) [inline storage/storage.go:80]: maybe it should be on top?

File: transaction/category.go [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package transaction
+
+const DefaultCategoryID = 1
```
Comment (flerka) [inline transaction/category.go:3]: Add comment here pls

File: transaction/transaction.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package transaction
+
+import "time"
+
+type Transaction struct {
```
Comment (flerka) [inline transaction/transaction.go:5]: Please add comments here

**Review by flerka — CHANGES_REQUESTED** [review]

File: transaction/category.go (hunk 4 above)
Comment (SuddenGunter) [reply transaction/category.go:4]: added

File: transaction/transaction.go (hunk 5 above)
Comment (SuddenGunter) [reply transaction/transaction.go:6]: added

Comment (SuddenGunter) [reply transaction/transaction.go:6]: added

File: storage/storage.go (hunk 3 above)
Comment (SuddenGunter) [reply storage/storage.go]: fixed

File: storage/storage.go (hunk 2 above)
Comment (SuddenGunter) [reply storage/storage.go]: fixed

File: mono/transaction.go (hunk 1 above)
Comment (SuddenGunter) [reply mono/transaction.go]: fixed

**Review by flerka — APPROVED** [review]

**Pull request merged**